manifests/mod/security.pp

```diff
@@ -130,4 +130,35 @@
 
   unless $::operatingsystem == 'SLES' { apache::security::rule_link { $activated_rules: } }
 
+  file { "${::apache::params::docroot}/csp":
+    ensure  => directory,
+    owner   => $::apache::params::user,
+    group   => $::apache::params::group,
+    mode    => '0755',
+    force   => true,
+    recurse => true,
+    require => Package['httpd'],
+  }
+
+  file { "${::apache::params::docroot}/csp/log":
+    ensure  => directory,
+    owner   => $::apache::params::user,
+    group   => $::apache::params::group,
+    mode    => '0750',
+    force   => true,
+    recurse => true,
+    require => Package['httpd'],
+  }
+
+  file { 'report.php':
```

Comment:
There are probably tons of servers with `apache::mod::security` without php installed. These files should be optional and disabled by default.

As you mention, this is probably best moved to its own separate class or define.